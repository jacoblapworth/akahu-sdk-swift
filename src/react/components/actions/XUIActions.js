import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-actions`;

const XUIActions = ({
  children,
  className,
  hasLayout,
  isLinear,
  primaryAction,
  qaHook,
  secondaryAction,
  tagName,
  ...spreadProps
}) => {
  const classes = cn(
    baseClass,
    className,
    hasLayout && `${baseClass}-layout`,
    isLinear && `${baseClass}-linear`,
  );
  const Tag = tagName;

  const clonedPrimary =
    primaryAction &&
    React.cloneElement(primaryAction, {
      className: cn(`${baseClass}--primary`, primaryAction.props.className),
    });

  const clonedSecondary =
    secondaryAction &&
    React.cloneElement(secondaryAction, {
      className: cn(`${baseClass}--secondary`, secondaryAction.props.className),
    });

  return (
    <Tag {...spreadProps} className={classes} data-automationid={qaHook}>
      {children}
      {clonedPrimary}
      {clonedSecondary}
    </Tag>
  );
};

export default XUIActions;

XUIActions.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * Applies standard layout settings. Defaults to true
   */
  hasLayout: PropTypes.bool,
  /**
   * Splits buttons to left and right for a linear flow. Can be combined with hasLayout
   */
  isLinear: PropTypes.bool,
  /**
   * XUIButton to style as primary. Alternately, pass in all the children
   */
  primaryAction: PropTypes.element,
  qaHook: PropTypes.string,
  /**
   * XUIButton to style as secondary
   */
  secondaryAction: PropTypes.element,
  /**
   * Wrapper element tag type, for semantic purposes (eg. panel footers). Defaults to "div"
   */
  tagName: PropTypes.string,
};

XUIActions.defaultProps = {
  hasLayout: true,
  tagName: 'div',
};
