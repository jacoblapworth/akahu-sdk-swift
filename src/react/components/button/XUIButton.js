import React from 'react';
import PropTypes from "prop-types";
import cn from 'classnames';
import XUILoader from '../loader/XUILoader';
import { VariantClassNames, SizeClassNames, ButtonTypes } from './private/constants';

/**
 * Returns true if the button is a borderless variant
 *
 * @private
 * @param {string} variant - The button variant
 * @return {boolean} True if is a borderless button
 */
const isBorderlessVariant = variant => {
	return variant.indexOf('borderless') > -1;
}

/**
 * Returns true if the button is an icon variant
 *
 * @private
 * @param {string} variant - The button variant
 * @return {boolean} True if is an icon button
 */
const isIconVariant = variant => {
	return variant.indexOf('icon') > -1;
}

/**
 * Returns a class name for the button depending on the button variant string given. Will return
 * undefined if no matching variant is given.
 *
 * @private
 * @param {string} variant - The button variant
 * @return {string} The variant specific class name
 */
const getVariantClass = variant => {
	return VariantClassNames.hasOwnProperty(variant) ? VariantClassNames[variant] : 'xui-button-standard';
};

/**
 * Replaces any href of `#` or undefined with `javascript:void(0)`. Else returns the passed href.
 *
 * @private
 * @param {string} href - A given link's href
 * @return {string} The href that will be assigned to a link
 */
const getHref = href => (!href || href === '#') ? 'javascript:void(0)' : href;

/**
 * KeyPress handler which will dispatch a click event when the space bar is pressed.
 *
 * @private
 * @param {KeyboardEvent} event
 * @param {Object} props
 */
function handleSpacebarAsClick(event, props) {
	if (props.isDisabled || props.isLoading) {
		let shouldClick;
		if (event.key) {
			shouldClick = event.key === ' ' || event.key === 'Spacebar';
		} else {
			shouldClick = (event.keyCode || event.which) === 32;
		}

		if (shouldClick) {
			// Clicking the space bar causes scrolling by default.  No bueno for a button.
			event.preventDefault();

			// A native event needs to be dispatched to ensure that all
			// browsers will follow the link.
			// Generate a click event with the latest API, if possible.
			// Use document.createEvent for IE 11.
			let clickEvent;
			if (typeof window.Event === 'function') {
				clickEvent = new MouseEvent('click', {
					bubbles: true,
					cancelable: true,
					view: window,
					detail: 0,
					screenX: 0,
					screenY: 0,
					clientX: 0,
					clientY: 0,
					ctrlKey: false,
					altKey: false,
					shiftKey: false,
					metaKey: false,
					button: 0,
					relatedTarget: null
				});
			} else {
				clickEvent = document.createEvent('MouseEvents');
				// If you're seeing a line through initMouseEvent because WebStorm, read the comment above the if statement
				clickEvent.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
			}
			event.target.dispatchEvent(clickEvent);
		}
	}
}

/**
 * Links styled as buttons require a specific set of attributes to ensure proper functionality and accessibility.
 * This function ensures that those props will be added to the element.
 *
 * @private
 * @param {Object} props
 * @param {Object} elementProps
 */
const setupLinkProps = (props, elementProps) => {
	// If this is just a plain link styled to be a button, the button role is not needed.  However, if this is a link
	// which is styled to look AND function like a button, then we'll need the role.
	if (!props.href || props.onClick) {
		elementProps.role = 'button';
	}
	elementProps.onKeyPress = function(e) {
		handleSpacebarAsClick(e, props);
	};
	elementProps.href = getHref(props.href);
	elementProps.target = props.target;
	elementProps.rel = props.rel;

	if (props.isExternalLink) {
		elementProps.rel = (elementProps.rel ? elementProps.rel + ' ' : '') + 'external noopener noreferrer'
	}

	if (props.isDisabled || props.isLoading) {
		elementProps['aria-disabled'] = true;
	}
};

/**
 * Attempt to focus the root DOM node of the given component, if the DOM node exists
 *
 * @private
 * @param {XUIButton} button
 */
const focusRootNode = button => button.rootNode != null && button.rootNode.focus();

export default class XUIButton extends React.Component {
	focus() {
		focusRootNode(this);
		// Apparently there are times when calling focus won't actually do it.  I think
		// React's getting in the way, but I'm not sure yet....
		if (this.rootNode !== document.activeElement) {
			setTimeout(focusRootNode, 0, this);
		}
	}

	hasFocus() {
		return this.rootNode == null ? false : this.rootNode.contains(document.activeElement);
	}

	render () {
		const xuiButton = this;
		const {
			type,
			variant,
			onClick,
			isDisabled,
			isLoading,
			children,
			className,
			onKeyDown,
			tabIndex,
			size,
			isGrouped,
			qaHook,
			isExternalLink,
			href,
			target,
			rel,
			isLink,
			isInverted,
			retainLayout,
			minLoaderWidth,
			...spreadProps
		} = xuiButton.props;
		const ElementType = isLink ? 'a' : 'button';
		const variantClass = getVariantClass(variant);
		const buttonDisabled = isDisabled || isLoading;
		let buttonChildren = children;

		const loader = (
			<XUILoader
				key={retainLayout && isLoading ? 'button-loader' : null}
				retainLayout={retainLayout}
				size="small"
				defaultLayout={false}
				className="xui-button--loader" />
		);

		if (retainLayout && isLoading) {
			buttonChildren = [
				<div className="xui-u-hidden-content" key='button-children'>{children}</div>,
				loader
			]
		} else if (isLoading) {
			buttonChildren = loader;
		}

		const buttonClassNames = cn('xui-button', className, variantClass, SizeClassNames[size], {
			'xui-button-is-disabled': isDisabled,
			'xui-button-grouped': isGrouped,
			'xui-button-inverted': isInverted && !isBorderlessVariant(variantClass) && !isIconVariant(variantClass),
			'xui-button-borderless-inverted': isInverted && isBorderlessVariant(variantClass) && !isIconVariant(variantClass),
			'xui-button-min-loader-width': minLoaderWidth
		});

		const clickHandler = function() {
			if (isLink && buttonDisabled) {
				event.preventDefault();
			} else if (onClick) {
				onClick.call(xuiButton, ...arguments);
			}
		};

		// Standard props for all element types
		const elementProps = {
			...spreadProps,
			onClick: clickHandler,
			onKeyDown: buttonDisabled ? null : onKeyDown,
			disabled: buttonDisabled,
			className: buttonClassNames,
			tabIndex: buttonDisabled ? -1 : tabIndex
		};

		// Element type specific props
		if (isLink) {
			setupLinkProps({ href, onClick, isExternalLink, target, rel, isDisabled, isLoading }, elementProps);
		} else {
			elementProps.type = type;
		}

		return (
			<ElementType ref={n => xuiButton.rootNode = n} {...elementProps} data-automationid={qaHook}>
				{buttonChildren}
			</ElementType>
		);
	}
}

XUIButton.propTypes = {
	className: PropTypes.string,
	children: PropTypes.node,
	qaHook: PropTypes.string,

	/** Determines if the button is disabled or not. */
	isDisabled: PropTypes.bool,

	/** If true, sets appropriate `rel` values to prevent new page from having access to `window.opener`. Should be used for links pointing at external sites. */
	isExternalLink: PropTypes.bool,

	/** If true, shows a loader inside the button and also disables the button to prevent clicking. Can be used in conjunction with isDisabled (which also provides a disabled class)  */
	isLoading: PropTypes.bool,

	/** If this button is part of a parent button group */
	isGrouped: PropTypes.bool,

	/** A keydown event handler for the button */
	onKeyDown: PropTypes.func,

	/** Bind a function to fire when the button is clicked */
	onClick: PropTypes.func,

	/** Determines the styling variation to apply: `standard`, `primary`, `create`, `negative`, `link`, 'borderless-standard', 'borderless-primary', 'borderless-create', 'borderless-negative', 'borderless-negative', 'icon', 'icon-large', 'icon-inverted', 'icon-inverted-large' or `unstyled`. */
	variant: PropTypes.oneOf(Object.keys(VariantClassNames)),

	/** Modifier for the size of the button. `small`, `full-width`, or `full-width-layout`. */
	size: PropTypes.oneOf(Object.keys(SizeClassNames)),

	/** Whether or not to render this button using an <a> tag */
	isLink: PropTypes.bool,

	/** The type attribute of this button. `submit`, `button`, or `reset`. */
	type: PropTypes.oneOf(Object.keys(ButtonTypes).map(type => ButtonTypes[type])), // Can't use Object.values without polyfilling for older supported browsers

	/** The `href` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
	href: PropTypes.string,

	/** The `rel` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
	rel: PropTypes.string,

	/** The HTML tabIndex attribute value */
	tabIndex: PropTypes.number,

	/** The `target` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
	target: PropTypes.string,

	/** The `title` attribute */
	title: PropTypes.string,

	/** Applies inverted styling */
	isInverted: PropTypes.bool,

	/** When used with `isLoading` this allows the button to retain children width */
	retainLayout: PropTypes.bool,

	/** Use this to specify a min width on the button, when you want to swap to loading states */
	minLoaderWidth: PropTypes.bool
};

XUIButton.defaultProps = {
	tabIndex: 0,
	type: ButtonTypes.button,
	variant: 'standard',
	isLink: false,
	isDisabled: false,
	isExternalLink: false,
	isGrouped: false,
	isLoading: false,
	retainLayout: true
};
