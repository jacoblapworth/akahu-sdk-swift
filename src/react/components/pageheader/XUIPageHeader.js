import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';
import { observe, unobserve } from '../helpers/containerQuery';
import WidthContext from '../../contexts/WidthContext';
import shouldRender from '../helpers/shouldRender';

const baseClass = `${ns}-pageheading`;

export default class XUIPageHeader extends PureComponent {
  _area = React.createRef();

  state = {};

  componentDidMount() {
    this._area.current && observe(this);
  }

  componentWillUnmount() {
    this._area.current && unobserve(this);
  }

  render() {
    const {
      qaHook,
      title,
      tabs,
      breadcrumb,
      className,
      children,
      actions,
      hasLayout,
      contentClassName,
      tags,
      secondary,
      supplementary,
      ...spreadProps
    } = this.props;

    const classes = cn(className, baseClass);
    const layoutClass = hasLayout ? `${baseClass}--content-layout` : '';
    const divClasses = cn(
      `${baseClass}--content`,
      !tabs && `${baseClass}--content-no-tabs`,
      layoutClass,
      contentClassName,
    );
    const rightClasses = cn(
      `${baseClass}--rightcontent`,
      !tabs && `${baseClass}--rightcontent-no-tabs`,
    );

    const clonedBreadcrumb =
      breadcrumb &&
      React.cloneElement(breadcrumb, {
        // Add the necessary pagehead class to the provided breadcrumb.
        className: cn(breadcrumb.props.className, `${baseClass}--breadcrumbtrail`),
      });
    const clonedTabs =
      tabs &&
      React.cloneElement(tabs, {
        // Make sure this is horizontal in the pagehead
        isHorizontal: true,
        className: cn(tabs.props.className, `${baseClass}--tabs`),
      });

    const titleTags = ([title, secondary, supplementary].some(shouldRender) || tags) && (
      <div
        className={cn(`${baseClass}--titlewrapper`, tags && `${baseClass}--titlewrapper-has-tags`)}
      >
        {shouldRender(title) && (
          <h1 className={`${baseClass}--title`} data-automationid={qaHook && `${qaHook}--title`}>
            {title}
          </h1>
        )}
        {shouldRender(secondary) && (
          <div
            className={`${baseClass}--secondarytitle`}
            data-automationid={qaHook && `${qaHook}--secondarytitle`}
          >
            {secondary}
          </div>
        )}
        {tags && <div className={`${baseClass}--tags`}>{tags}</div>}
        {shouldRender(supplementary) && (
          <div
            className={`${baseClass}--supplementarytext`}
            data-automationid={qaHook && `${qaHook}--supplementarytext`}
          >
            {supplementary}
          </div>
        )}
      </div>
    );
    const leftContent = (titleTags || clonedBreadcrumb) && (
      <div className={`${baseClass}--leftcontent`}>
        {clonedBreadcrumb}
        {titleTags}
      </div>
    );
    const rightContent = (clonedTabs || actions) && (
      <div className={rightClasses}>
        {clonedTabs}
        {actions && <div className={`${baseClass}--actions`}>{actions}</div>}
      </div>
    );
    return (
      <WidthContext.Provider value={this.state}>
        <header {...spreadProps} className={classes} data-automationid={qaHook}>
          <div className={divClasses} ref={this._area}>
            {leftContent}
            {children}
            {rightContent}
          </div>
        </header>
      </WidthContext.Provider>
    );
  }
}

XUIPageHeader.propTypes = {
  /**
   * Components or html to be right-aligned in the pageheading
   */
  actions: PropTypes.node,
  /**
   * Instantiated breadcrumb component. Use in conjunction with a title.
   */
  breadcrumb(props, propName) {
    if (props[propName] && !props.title) {
      return new Error('Breadcrumb in a page header is to be used with a title');
    }
    return null;
  },
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * CSS class(es) to add to the the pageheading--content element. xui-page-width-standard
   * would go here
   */
  contentClassName: PropTypes.string,
  /**
   * Applies default layout styling.
   */
  hasLayout: PropTypes.bool,
  qaHook: PropTypes.string,
  /**
   * Secondary title
   */
  secondary: PropTypes.node,
  /**
   * Supplementary text to appear after the headings and tags
   */
  supplementary: PropTypes.node,
  /**
   * Horizontal picklist to act as tabs. XUIPicklist
   */
  tabs: PropTypes.element,
  /**
   * Array of XUITags
   */
  tags: PropTypes.arrayOf(PropTypes.element),
  /**
   * Title text or node. Should almost certainly be present.
   */
  title: PropTypes.node,
};

XUIPageHeader.defaultProps = {
  hasLayout: true,
};
