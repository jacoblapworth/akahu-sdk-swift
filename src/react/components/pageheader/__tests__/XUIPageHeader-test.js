import React from 'react';
import XUIPageHeader from '../XUIPageHeader';
import XUIBreadcrumb from '../XUIBreadcrumb';
import XUIPicklist from '../../picklist/Picklist';
import XUIPickitem from '../../picklist/Pickitem';
import XUIButton from '../../button/XUIButton';
import XUIActions from '../../actions/XUIActions';
import WidthContext from '../../../contexts/WidthContext';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('<XUI PageHeader and Breadcrumb/>', () => {
  const qaHook = 'qaHook';
  const primary = <XUIButton>Primary</XUIButton>;
  const secondary = <XUIButton>Secondary</XUIButton>;
  const tabs = (
    <XUIPicklist>
      <XUIPickitem id="1">Tab 1</XUIPickitem>
      <XUIPickitem id="2" isSelected={true}>
        Tab 2
      </XUIPickitem>
      <XUIPickitem id="3">This is tab 3</XUIPickitem>
    </XUIPicklist>
  );
  const actions = <XUIActions primary={primary} secondary={secondary} />;
  const bcObj = [<span
    className="testy-mctesterson"
    key="1"
    onClick={() => alert('hello')}
    onKeyDown={() => {}}
    role="link"
    tabIndex="0"
  >
    hello
  </span>, { label: 'hiya', href: '#2' }, { label: 'yo' }];
  const bcObj2 = [{ label: 'Settings', href: '#1' }, { label: 'Edit', href: '#2' }];
  const bcObj3 = [
    { label: 'Settings', href: '#1' },
    { label: 'Edit', href: '#2' },
    { label: 'Invite', href: '#3' },
  ];
  const exampleBreadcrumb = <XUIBreadcrumb breadcrumbs={bcObj} qaHook={qaHook} />;
  it('renders the simplest pageHeader with no extra settings passed', () => {
    const testPageHeader = renderer.create(<XUIPageHeader title="Testing ðŸ’©" />);
    expect(testPageHeader).toMatchSnapshot();
  });
  it('renders pageHeader without default layout', () => {
    const wrapper = mount(<XUIPageHeader title="Testing ðŸ’©" hasLayout={false} />);
    expect(wrapper.find('.xui-pageheading--content-layout').length).toBe(0);
  });
  it('renders extra pageHeader classes that are passed in', () => {
    const testPageHeader = renderer.create(
      <XUIPageHeader title="Testing ðŸ’©" className="testClass" contentClassName="testClassTwo" />,
    );
    expect(testPageHeader).toMatchSnapshot();
  });
  it('renders pageHeader containing Actions', () => {
    const wrapper = mount(<XUIPageHeader actions={actions} />);
    expect(wrapper.find('.xui-pageheading--actions').length).toBe(1);
  });
  it('renders pageHeader with title and Actions', () => {
    const wrapper = mount(<XUIPageHeader title="Testing ðŸ’©" actions={actions} />);
    expect(wrapper.find('.xui-pageheading--actions').length).toBe(1);
    expect(wrapper.find('.xui-pageheading--title').length).toBe(1);
  });
  it('renders pageHeader with title and tabs and Actions', () => {
    const wrapper = mount(<XUIPageHeader title="Testing ðŸ’©" tabs={tabs} actions={actions} />);
    expect(wrapper.find('.xui-pageheading--actions').length).toBe(1);
    expect(wrapper.find('.xui-pageheading--title').length).toBe(1);
    expect(wrapper.find('ul.xui-pageheading--tabs').length).toBe(1);
  });
  it('renders pageHeader with Breadcrumb and Actions', () => {
    const wrapper = mount(
      <XUIPageHeader breadcrumb={exampleBreadcrumb} actions={actions} title="Test" />,
    );
    expect(wrapper.find('.xui-pageheading--actions').length).toBe(1);
    expect(wrapper.find('ol.xui-breadcrumbs').length).toBe(1);
  });
  it('errors when passed a Breadcrumb with no title', () => {
    expect(() => renderer.create(<XUIPageHeader breadcrumb={exampleBreadcrumb} />)).toThrow();
  });
  it('renders pageHeader containing Breadcrumb from nodes', () => {
    const bcNodeObj = [
      <span role="link" onClick={() => alert('hello')} key="1">
        hello
      </span>,
      <span role="link" onClick={() => alert('hiya')} key="2">
        hello
      </span>,
      <span key="3">yo</span>,
    ];
    const exampleNodeBreadcrumb = <XUIBreadcrumb breadcrumbs={bcNodeObj} />;
    const testPageHeader = renderer.create(
      <XUIPageHeader breadcrumb={exampleNodeBreadcrumb} title="Test" />,
    );
    expect(testPageHeader).toMatchSnapshot();
  });
  it('renders pageHeader containing tabs', () => {
    const wrapper = mount(<XUIPageHeader tabs={tabs} />);
    expect(wrapper.find('ul.xui-pageheading--tabs').length).toBe(1);
    expect(wrapper.find('.xui-pageheading--title').length).toBe(0);
  });
  it('renders pageHeader containing title and tabs', () => {
    const wrapper = mount(<XUIPageHeader title="Testing ðŸ’©" tabs={tabs} />);
    expect(wrapper.find('ul.xui-pageheading--tabs').length).toBe(1);
  });
  it('renders pageHeader with tabs, title, and Breadcrumb when all are passed', () => {
    const wrapper = mount(
      <XUIPageHeader title="Testing ðŸ’©" tabs={tabs} breadcrumb={exampleBreadcrumb} />,
    );
    expect(wrapper.find('ul.xui-pageheading--tabs').length).toBe(1);
    expect(wrapper.find('.xui-pageheading--title').length).toBe(1);
    expect(wrapper.find('ol.xui-breadcrumbs').length).toBe(1);
  });
  it('renders pageHeader and breadcrumb with automation id when qaHook prop is passed in', () => {
    const wrapper = renderer.create(
      <XUIPageHeader qaHook={qaHook} breadcrumb={exampleBreadcrumb} title="Test" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders compact Breadcrumb, when at a width context below the specified size and breadcrumb length > 2', () => {
    const contextualBc = renderer.create(
      <WidthContext.Provider value={{ medium: false, small: true }}>
        <XUIBreadcrumb breadcrumbs={bcObj3} swapAtBreakpoint="medium" />
      </WidthContext.Provider>,
    );
    expect(contextualBc).toMatchSnapshot();
  });
  it('does not condense Breadcrumb, if context is too wide', () => {
    const contextualBc = renderer.create(
      <WidthContext.Provider value={{ medium: true, small: true }}>
        <XUIBreadcrumb breadcrumbs={bcObj3} swapAtBreakpoint="medium" />
      </WidthContext.Provider>,
    );
    expect(contextualBc).toMatchSnapshot();
  });
  it('does not condense Breadcrumb, if breadcrumb length < 3', () => {
    const contextualBc = renderer.create(
      <WidthContext.Provider value={{ medium: false, small: true }}>
        <XUIBreadcrumb breadcrumbs={bcObj2} swapAtBreakpoint="medium" />
      </WidthContext.Provider>,
    );
    expect(contextualBc).toMatchSnapshot();
  });
});
