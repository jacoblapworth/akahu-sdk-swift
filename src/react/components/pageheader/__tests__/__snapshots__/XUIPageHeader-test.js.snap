// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XUI PageHeader and BreadcrumbTrail/> does not condense Breadcrumb, if context is too wide 1`] = `
<ol
  className="xui-breadcrumbtrail"
>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#1"
    >
      Settings
    </a>
  </li>
  <li
    className="xui-breadcrumb-arrow"
  >
    <div
      className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
    >
      <svg
        className="xui-icon xui-icon-rotate-270"
        focusable="false"
        height={4}
        viewBox="0 0 7 4"
        width={7}
      >
        <path
          d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
          role="presentation"
        />
      </svg>
    </div>
  </li>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#2"
    >
      Edit
    </a>
  </li>
  <li
    className="xui-breadcrumb-arrow"
  >
    <div
      className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
    >
      <svg
        className="xui-icon xui-icon-rotate-270"
        focusable="false"
        height={4}
        viewBox="0 0 7 4"
        width={7}
      >
        <path
          d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
          role="presentation"
        />
      </svg>
    </div>
  </li>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#3"
    >
      Invite
    </a>
  </li>
</ol>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> does not condense XUIBreadcrumbTrail, if breadcrumb length < 3 1`] = `
<ol
  className="xui-breadcrumbtrail"
>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#1"
    >
      Settings
    </a>
  </li>
  <li
    className="xui-breadcrumb-arrow"
  >
    <div
      className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
    >
      <svg
        className="xui-icon xui-icon-rotate-270"
        focusable="false"
        height={4}
        viewBox="0 0 7 4"
        width={7}
      >
        <path
          d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
          role="presentation"
        />
      </svg>
    </div>
  </li>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#2"
    >
      Edit
    </a>
  </li>
</ol>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders compact XUIBreadcrumbTrail, when at a width context below the specified size and breadcrumb length > 2 1`] = `
<ol
  className="xui-breadcrumbtrail"
>
  <li
    className="xui-breadcrumb"
  >
    <div
      className=""
      data-ref="toggled-wrapper"
      role="presentation"
    >
      <button
        aria-activedescendant={null}
        aria-haspopup="listbox"
        className="xui-button xui-breadcrumb--link xui-breadcrumb--dropdowntrigger"
        disabled={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        type="button"
      >
        ...
      </button>
    </div>
  </li>
  <li
    className="xui-breadcrumb-arrow"
  >
    <div
      className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
    >
      <svg
        className="xui-icon xui-icon-rotate-270"
        focusable="false"
        height={4}
        viewBox="0 0 7 4"
        width={7}
      >
        <path
          d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
          role="presentation"
        />
      </svg>
    </div>
  </li>
  <li
    className="xui-breadcrumb"
  >
    <a
      className="xui-breadcrumb--link"
      href="#3"
    >
      Invite
    </a>
  </li>
</ol>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders extra pageHeader classes that are passed in 1`] = `
<header
  className="testClass xui-pageheading"
>
  <div
    className="xui-pageheading--content xui-pageheading--content-no-tabs xui-pageheading--content-layout testClassTwo"
  >
    <div
      className="xui-pageheading--leftcontent"
    >
      <div
        className="xui-pageheading--titlewrapper"
      >
        <h1
          className="xui-pageheading--title"
        >
          Testing ðŸ’©
        </h1>
      </div>
    </div>
  </div>
</header>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders pageHeader and breadcrumbtrail with automation id when qaHook prop is passed in 1`] = `
<header
  className="xui-pageheading"
  data-automationid="qaHook"
>
  <div
    className="xui-pageheading--content xui-pageheading--content-no-tabs xui-pageheading--content-layout"
  >
    <div
      className="xui-pageheading--leftcontent"
    >
      <ol
        className="xui-pageheading--breadcrumbtrail xui-breadcrumbtrail"
        data-automationid="qaHook"
      >
        <li
          className="xui-breadcrumb"
        >
          <span
            className="testy-mctesterson xui-breadcrumb--link"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex="0"
          >
            hello
          </span>
        </li>
        <li
          className="xui-breadcrumb-arrow"
        >
          <div
            className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
          >
            <svg
              className="xui-icon xui-icon-rotate-270"
              focusable="false"
              height={4}
              viewBox="0 0 7 4"
              width={7}
            >
              <path
                d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
                role="presentation"
              />
            </svg>
          </div>
        </li>
        <li
          className="xui-breadcrumb"
        >
          <a
            className="xui-breadcrumb--link"
            href="#2"
          >
            hiya
          </a>
        </li>
        <li
          className="xui-breadcrumb-arrow"
        >
          <div
            className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
          >
            <svg
              className="xui-icon xui-icon-rotate-270"
              focusable="false"
              height={4}
              viewBox="0 0 7 4"
              width={7}
            >
              <path
                d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
                role="presentation"
              />
            </svg>
          </div>
        </li>
        <li
          className="xui-breadcrumb xui-breadcrumb-no-link"
        >
          yo
        </li>
      </ol>
      <div
        className="xui-pageheading--titlewrapper"
      >
        <h1
          className="xui-pageheading--title"
          data-automationid="qaHook--title"
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</header>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders pageHeader containing XUIBreadcrumbTrail from nodes 1`] = `
<header
  className="xui-pageheading"
>
  <div
    className="xui-pageheading--content xui-pageheading--content-no-tabs xui-pageheading--content-layout"
  >
    <div
      className="xui-pageheading--leftcontent"
    >
      <ol
        className="xui-pageheading--breadcrumbtrail xui-breadcrumbtrail"
      >
        <li
          className="xui-breadcrumb"
        >
          <span
            className="xui-breadcrumb--link"
            onClick={[Function]}
            role="link"
          >
            hello
          </span>
        </li>
        <li
          className="xui-breadcrumb-arrow"
        >
          <div
            className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
          >
            <svg
              className="xui-icon xui-icon-rotate-270"
              focusable="false"
              height={4}
              viewBox="0 0 7 4"
              width={7}
            >
              <path
                d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
                role="presentation"
              />
            </svg>
          </div>
        </li>
        <li
          className="xui-breadcrumb"
        >
          <span
            className="xui-breadcrumb--link"
            onClick={[Function]}
            role="link"
          >
            hello
          </span>
        </li>
        <li
          className="xui-breadcrumb-arrow"
        >
          <div
            className="xui-breadcrumb--icon xui-iconwrapper xui-iconwrapper-medium"
          >
            <svg
              className="xui-icon xui-icon-rotate-270"
              focusable="false"
              height={4}
              viewBox="0 0 7 4"
              width={7}
            >
              <path
                d="M3.5 2.588L.75 0 0 .706 3.5 4 7 .706 6.25 0z"
                role="presentation"
              />
            </svg>
          </div>
        </li>
        <li
          className="xui-breadcrumb"
        >
          <span
            className="xui-breadcrumb--link"
          >
            yo
          </span>
        </li>
      </ol>
      <div
        className="xui-pageheading--titlewrapper"
      >
        <h1
          className="xui-pageheading--title"
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</header>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders pageHeader with main title, secondary title, and supplementary text 1`] = `
<header
  className="xui-pageheading"
>
  <div
    className="xui-pageheading--content xui-pageheading--content-no-tabs xui-pageheading--content-layout"
  >
    <div
      className="xui-pageheading--leftcontent"
    >
      <div
        className="xui-pageheading--titlewrapper"
      >
        <h1
          className="xui-pageheading--title"
        >
          Testing ðŸ’©
        </h1>
        <div
          className="xui-pageheading--secondarytitle"
        >
          Testing ðŸ’©
        </div>
        <div
          className="xui-pageheading--supplementarytext"
        >
          And more testing ðŸ’©
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`<XUI PageHeader and BreadcrumbTrail/> renders the simplest pageHeader with no extra settings passed 1`] = `
<header
  className="xui-pageheading"
>
  <div
    className="xui-pageheading--content xui-pageheading--content-no-tabs xui-pageheading--content-layout"
  >
    <div
      className="xui-pageheading--leftcontent"
    >
      <div
        className="xui-pageheading--titlewrapper"
      >
        <h1
          className="xui-pageheading--title"
        >
          Testing ðŸ’©
        </h1>
      </div>
    </div>
  </div>
</header>
`;
